app/serializers/theme_serializer.rb

```diff
@@ -55,6 +57,14 @@ def parent_themes
     object.parent_themes
   end
 
+  def base_palette
+    ColorScheme.base
+  end
+
+  def include_base_palette?
+    object.color_scheme_id.blank? && object.owned_color_palette.blank?
```

Comment:
I don't seem to see `owned_color_palette` being eager loaded anywhere so this may result in an N+1 queries problem.

Reply:
(having said that, I think this serializer should never be used to render a list of themes --- it's sends way too much data for any list. In the new themes and components listing pages, we have smaller serializers that include only the data needed for these respective pages)

Reply:
> (having said that, I think this serializer should never be used to render a list of themes --- it sends way too much data for any list. In the new themes and components listing pages, we have smaller serializers that include only the data needed for these respective pages)

I see so does that means this controller action will eventually be removed? https://github.com/discourse/discourse/blob/4e4ff41d19a096a114b420516cb18d06c642178e/app/controllers/admin/themes_controller.rb#L178